engine/src/property.cpp

```diff
@@ -1237,34 +1237,49 @@ bool MCProperty::resolveprop(MCExecContext& ctxt, Properties& r_which, MCNameRef
         if (!ctxt  . EvalExprAsStringRef(destvar, EE_PROPERTY_BADEXPRESSION, &t_string))
             return false;
 
-        MCScriptPoint sp(*t_string);
+        // AL-2015-08-27: [[ Bug 15798 ]] Parse into index and property name before determining
+        //  if this is a custom property or not (otherwise things like textStyle["bold"] are
+        //  interpreted as custom properties).
+        MCAutoStringRef t_icarray, t_property_name;
+        uindex_t t_offset, t_end_offset;
+        // SN-2014-08-08: [[ Bug 13135 ]] Targetting a custom var should resolve the index AND suppress
+        //  it from the initial name
+        if (MCStringFirstIndexOfChar(*t_string, '[', 0, kMCStringOptionCompareExact, t_offset) &&
+            MCStringLastIndexOfChar(*t_string, ']', UINDEX_MAX, kMCStringOptionCompareExact, t_end_offset) &&
+            t_end_offset == MCStringGetLength(*t_string) - 1)
+        {
+            MCStringCopySubstring(*t_string, MCRangeMake(0, t_offset), &t_property_name);
+            
+            // AL-2015-08-27: [[ Bug 15798 ]] If the index is quoted, we don't want to include the
+            //  quotes in the index name.
+            if (MCStringGetCodepointAtIndex(*t_string, t_offset + 1) == '"' &&
+                MCStringGetCodepointAtIndex(*t_string, t_end_offset - 1) == '"')
+            {
+                t_offset++;
+                t_end_offset--;
+            }
+            
+            MCStringCopySubstring(*t_string, MCRangeMake(t_offset + 1, t_end_offset - t_offset - 1), &t_icarray);
+        }
+        else
+            t_property_name = *t_string;
+        
+        if (*t_icarray != nil)
+        /* UNCHECKED */ MCNameCreate(*t_icarray, t_index_name);
```

Comment:
@livecodeali: Is that true in this case?

The function returns false if it fails to resolve the property. If the callers take 'false' to mean 'an error has been thrown' then a function returning false due to a memory error should return false. If the callers use a 'false' error return to work out whether to do other processing, then it becomes an issue.

In this case, EvalExpr is called which will push an error onto the error stack if it fails, so I suspect the callers just throw an error if the function returns false rather than doing different processing.